runtime.js

```diff
@@ -15,10 +17,23 @@ function checkProcessArgv (moduleName) {
 let preloadModules
 function checkPreloadModules (moduleName) {
   /* c8 ignore start */
-  // nullish needed for non Node.js runtime
+  // coverage - nullish needed for non Node.js runtime
   preloadModules ??= (process._preload_modules ?? [])
+
+  // coverage - TS specific
+  if (preloadModules.includes(moduleName)) {
+    return true
+  }
+
+  const modulePath = path.join(process.cwd(), 'node_modules', moduleName)
+  if (Object.keys(require.cache).some(k => k.startsWith(modulePath))) {
+    preloadModules.push(moduleName)
```

Comment:
I still discourage mutating `preloadModules`, as you may mutate the `process._preload_modules`.

Reply:
https://github.com/fastify/fastify-autoload/pull/406/files#diff-620d8f84607e4a3c91579ab868c89080849559f6533315da0c0ec092f9899a11R21

Reply:
I think I am gonna do a PR specific for immutability as there is a lot of chance that this PR will close. I don't feel confident with it.